### DISCUSSION TITLE
Error when using code with React Markdown in Next.js with TypeScript

### DISCUSSION THREAD

**Discussion by pavinduLakshan** [remarkjs/remark#813] [Q&A] [answered]
I'm trying to add react-markdown to my next.js app, but it complains about some TS-type problems and I'm not able to resolve it myself.

All details are documented here.   
https://stackoverflow.com/questions/68799286/error-when-using-partialnormalcomponents-specialcomponents-in-nextjs-with-ty

It will be a great favor done if you can answer this question for me. Thank you very much for your time.

**Comment by wooorm** [comment]
This is most likely a typescript error and unrelated to next.
I would recommend importing `Components` from `react-markdown` instead of diving into the internal types.
But otherwise I’m not sure what the problem is exactly.

**Comment by ChristianMurphy** [comment, marked as answer]
@pavinduLakshan try inlining `component` to ensure it works first, for example https://codesandbox.io/s/react-markdown-typescript-syntax-highlight-00uz6
Then, if you want to extract it again, consider importing `import { Options } from 'react-markdown'` (for version 7+), and access the components type with `Options['components']` to type the `components` object.
